controllers/replicationdestination.go

```diff
@@ -220,3 +221,121 @@ func (r *VolumeSnapshotRestoreReconciler) SetVSRStatus(log logr.Logger) (bool, e
+			}
+
+			// check for config accessMode, otherwise use source PVC accessMode
+			if spec == "DestinationAccessMode" {
+				repDestAccessMode = cm.Data["DestinationAccessMode"]
+				repDestAccessModeAM = []corev1.PersistentVolumeAccessMode{corev1.PersistentVolumeAccessMode(repDestAccessMode)}
+
+				// cm.Data is set but field not set
+			} else {
+				repDestAccessModeAM = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+			}
+		}
+
+		// no configmap data used
+	} else {
+
+		repDestStorageClass = vsr.Spec.VolumeSnapshotMoverBackupref.BackedUpPVCData.StorageClassName
+		repDestAccessModeAM = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+	}
+
+	repDestVolOptions.StorageClassName = &repDestStorageClass
+	repDestVolOptions.AccessModes = repDestAccessModeAM
+
+	return &repDestVolOptions, nil
+}
+
+func (r *VolumeSnapshotRestoreReconciler) configureRepDestResticVolOptions(vsr *volsnapmoverv1alpha1.VolumeSnapshotRestore, resticSecretName string,
+	cm *corev1.ConfigMap, capacity *resource.Quantity) (*volsyncv1alpha1.ReplicationDestinationResticSpec, error) {
+
+	if vsr == nil {
+		return nil, errors.New("nil vsr in configureRepDestResticVolOptions")
+	}
+
+	if capacity == nil {
+		return nil, errors.New("nil capacity in configureRepDestResticVolOptions")
+	}
+
+	repDestResticVolOptions := volsyncv1alpha1.ReplicationDestinationResticSpec{}
+	repDestResticVolOptions.Repository = resticSecretName
+
+	var repDestCacheStorageClass string
+	var repDestCaceheStorageClassPt *string
+
+	var repDestCacheAccessMode string
+
+	var repDestCacheCapacity string
+	var repDestCacheCapacityCp resource.Quantity
+
+	if cm != nil && cm.Data != nil {
+		for spec := range cm.Data {
+
+			// check for config cacheStorageClassName, otherwise use source PVC storageClass
+			if spec == "DestinationCacheStorageClassName" {
+				repDestCacheStorageClass = cm.Data["DestinationCacheStorageClassName"]
+				repDestCaceheStorageClassPt = &repDestCacheStorageClass
+
+				repDestResticVolOptions.CacheStorageClassName = repDestCaceheStorageClassPt
+			}
+
+			// check for config cacheAccessMode, otherwise use source PVC accessMode
+			if spec == "DestinationCacheAccessMode" {
+				repDestCacheAccessMode = cm.Data["DestinationCacheAccessMode"]
+				repDestResticVolOptions.CacheAccessModes = []corev1.PersistentVolumeAccessMode{corev1.PersistentVolumeAccessMode(repDestCacheAccessMode)}
+			}
+
+			// check for config cacheCapacity, otherwise use source PVC capacity
+			if spec == "DestinationCacheCapacity" {
+				repDestCacheCapacity = cm.Data["DestinationCacheCapacity"]
+				repDestCacheCapacityCp = resource.MustParse(repDestCacheCapacity)
+
+				repDestResticVolOptions.CacheCapacity = &repDestCacheCapacityCp
+			}
+		}
+	}
+
+	optionsSpec, err := r.configureRepDestVolOptions(vsr, capacity, cm)
+	if err != nil {
+		return nil, err
+	}
+
+	repDestResticVolOptions.ReplicationDestinationVolumeOptions = *optionsSpec
```

Comment:
In case shit happens.
```suggestion
	if optionsSpec != nil {
		repDestResticVolOptions.ReplicationDestinationVolumeOptions = *optionsSpec
	}
```